PyInstaller/depend/dylib.py

```diff
@@ -225,23 +225,28 @@ def search(self, libname):
 
     exclude_list = WinExcludeList(exclude_list)
 
+_seen_wine_dlls = set()  # Used for warning tracking in include_library()
+
 
 def include_library(libname):
     """
     Check if the dynamic library should be included with application or not.
     """
-    # For configuration phase we need to have exclude / include lists None so these checking is skipped and library gets
-    # included.
     if exclude_list:
         if exclude_list.search(libname) and not include_list.search(libname):
-            # Library is excluded and is not overriden by include list. It should be then excluded.
+            # Library is excluded and is not overriden by include list. It should be excluded.
             return False
-        else:
-            # Include library.
-            return True
-    else:
-        # By default include library.
-        return True
+
+    # If we are running under Wine and the library is a Wine built-in DLL, exclude it, regardless of include and
+    # exclude lists. If excluded, the frozen application will still run under Wine, but if it were included, it would
+    # likely cause problems under real Windows. So everything considered, we are better off excluding it.
+    if compat.is_win_wine and compat.is_wine_dll(libname):
+        if libname not in _seen_wine_dlls:
+            logger.warning("Excluding Wine built-in DLL: %s", libname)  # displayed only if DLL would have been included
```

Comment:
This doesn't say why having to exclude a Wine DLL is potentially hazardous. i.e. That if the application needed that DLL then the bundle is incomplete.

Reply:
The revised text hopefully better explains the dilemma. Or the lack of it, as it were.